learning_resources_search/serializers.py

```diff
@@ -158,14 +158,41 @@ def to_representation(self, data):
     "professional",
 ]
 
-CONTENT_FILE_AGGREGATIONS = ["topic", "content_category"]
+CONTENT_FILE_AGGREGATIONS = ["topic", "content_category", "platform", "offered_by"]
 
 
 class SearchRequestSerializer(serializers.Serializer):
     q = serializers.CharField(required=False, help_text="The search text")
     offset = serializers.IntegerField(required=False)
     limit = serializers.IntegerField(required=False)
     id = StringArrayField(required=False, child=serializers.CharField())  # noqa: A003
+    offered_by = StringArrayField(
+        required=False,
+        child=serializers.ChoiceField(choices=[e.value.lower() for e in OfferedBy]),
+    )
+    platform = StringArrayField(
+        required=False,
+        child=serializers.ChoiceField(choices=[e.value.lower() for e in PlatformType]),
+    )
+    topic = StringArrayField(required=False, child=serializers.CharField())
+
+    def validate_offered_by(self, data):
+        if data:
+            for offered_by_value in data.split(","):
+                if offered_by_value.lower() not in [e.value.lower() for e in OfferedBy]:
+                    msg = f"{offered_by_value} is not a valid option"
+                    raise serializers.ValidationError(msg)
+        return data
```

Comment:
If we modify StringListField a little, I believe we get all this validation for free. Currently we do not because `StringListField.to_internal_value` does not call its parent method, but with something like
```python
class StringArrayField(serializers.ListField):
    """
    Character separated ListField.
    """

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

    def to_internal_value(self, data):
        """
        Turn ["a", "b,c"] into ["a", "b", "c"]
        """
        normalized = ",".join(data).split(",")
        return super().to_internal_value(normalized)
```
the child serializer should handle the validation automatically.